### PR TITLE
Initial account management functionality

### PR DESCRIPTION
**Pull request by stepbeekio**
This includes a table listing all current accounts as well as user impersonation.

If it's still open, you should do this guy first: https://github.com/HappyValleyIO/springboard/pull/21

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -78,6 +82,7 @@ dependencies {
   testImplementation("org.springframework.boot:spring-boot-starter-test") {
     exclude group: "org.junit.vintage", module: "junit-vintage-engine"
   }
+  testImplementation("com.willowtreeapps.assertk:assertk-jvm:0.21")
```
Comment (hevb) [inline build.gradle:85]: What is willowtreeapps? Just the creator of assertk? I've never heard of it

File: src/main/resources/db/migration/V1__authentication.sql [hunk 2]
```diff
@@ -1,17 +1,18 @@
+CREATE EXTENSION IF NOT EXISTS "uuid-ossp";
```
Comment (hevb) [inline src/main/resources/db/migration/V1__authentication.sql:1]: What is a UUID-OSSP?

File: src/test/kotlin/org/springboard/UserServiceIntegrationTest.kt [hunk 3]
```diff
@@ -1,28 +1,53 @@
...
-  internal fun givenAnExistingUserThenLoginWithCorrectCredentialsResultsInAccount() {
-    val email = "stephen@getspringboard.dev"
-    val password = "SomePassword456!"
-    val created: UserCreationResult.CreatedUser =
-      userService.createUser(UserCreationRequest("Stephen the tester", email, password, "tester")) as UserCreationResult.CreatedUser
-    val queried = userService.login(email, password)
-
-    assertEquals("Assert logged in user is same as created user", created.account, queried)
-  }
+class UserServiceIntegrationTest(@Autowired val userService: UserService, @Autowired val userDetails: UserDetailsService) {
```
Comment (hevb) [inline src/test/kotlin/org/springboard/UserServiceIntegrationTest.kt:19]: I guess I thought this would extend the base integration test? - Why doesn't it?

File: src/test/kotlin/org/springboard/admin/AdminServiceTest.kt [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+import org.springboard.user.UserCreationResult
+import org.springboard.user.UserServiceImpl
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.boot.test.context.SpringBootTest
+import org.springframework.test.context.junit.jupiter.SpringExtension
+import java.util.*
+
+@ExtendWith(SpringExtension::class)
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.NONE)
+class AdminServiceTest {
```
Comment (hevb) [inline src/test/kotlin/org/springboard/admin/AdminServiceTest.kt:19]: Should service tests extend a baseServiceTest?

File: src/test/kotlin/org/springboard/user/InternalUserTest.kt [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+import assertk.assertThat
+import assertk.assertions.isEqualTo
+import assertk.assertions.isFalse
+import assertk.assertions.isTrue
+import org.junit.jupiter.api.Test
+import org.springboard.auth.AuthAuthority
+import java.time.Instant
+import java.util.*
+
```
Comment (hevb) [inline src/test/kotlin/org/springboard/user/InternalUserTest.kt:11]: extra spacing

File: src/webapp/src/templates/components/hint.peb [hunk 6]
```diff
@@ -1,3 +1,4 @@
+/**
```
Comment (hevb) [inline src/webapp/src/templates/components/hint.peb:1]: Why is this here Is there a **/ end to the comment somewhere This feels like it was accidentally commented out?

**Review by hevb — APPROVED** [review]
A few minor questions but I'm happy for you to self merge this after looking into them. 👍

File: build.gradle (hunk 1 above)
Comment (stepbeekio) [reply build.gradle]: It's just the creator of assertk (which is just a kotlin wrapper around assertj)

File: src/main/resources/db/migration/V1__authentication.sql (hunk 2 above)
Comment (stepbeekio) [reply src/main/resources/db/migration/V1__authentication.sql]: UUID-OSSP is a postgresql extension that we can enable that let's us generate UUID's directly in postgresl. So we can have

```sql
   customer_id UUID NOT NULL DEFAULT UUID()
```

File: src/test/kotlin/org/springboard/UserServiceIntegrationTest.kt (hunk 3 above)
Comment (stepbeekio) [reply src/test/kotlin/org/springboard/UserServiceIntegrationTest.kt]: Fixing now

File: src/test/kotlin/org/springboard/admin/AdminServiceTest.kt (hunk 4 above)
Comment (stepbeekio) [reply src/test/kotlin/org/springboard/admin/AdminServiceTest.kt:19]: I'm going to do BaseIntegrationTest for now. I don't want to have services tested in a particular way. I think they're only interesting if we're testing the whole Spring show.

File: src/webapp/src/templates/components/hint.peb (hunk 6 above)
Comment (stepbeekio) [reply src/webapp/src/templates/components/hint.peb:1]: Fixed.

File: src/test/kotlin/org/springboard/user/InternalUserTest.kt (hunk 5 above)
Comment (stepbeekio) [reply src/test/kotlin/org/springboard/user/InternalUserTest.kt:11]: Fixed.
